CONTRIBUTING.md

```diff
@@ -135,6 +135,30 @@ Most features will likely have most of the code in the `containerd-shim-wasm` pr
 
 A tip for developing a new feature is to implement it and test it with one runtime you are familiar with then add it to all the runtimes.  This makes it easier to test and iterate before making changes across all the runtimes.
 
+## Adding new shims
+
+We welcome new shims, though you can also host them in your own repositories as well and use the `coantainerd-shim-wasm` crate.  We recognize that the project is moving fast and having them in this repository can reduce the need for changes as well for discoverability.  
+
+Please open an issue before submitting a PR for discussion to make sure it is a good fit.  As a general rule, we want shims to be adopting WASI spec (this is after all called run*wasi* :-)). In the future we may require shims in the repository to pass WASI compliance tests when the standards mature more. See https://github.com/containerd/runwasi/issues/338 for more discussion.
+
+Since we are not experts in every runtime we also need a commitment from the runtime owners to contribute to maintenance of the shim.
+
+## Removing Shims
+
+This is a fast moving space, with lots of innovation happening and some shims may eventually need to be removed.
+
+A Shim implementation maybe subject to removal if:
+ - If a shim runtime has not been maintained for 6 months it will be subject for removal
+ - If required changes to the runtime can't be merged or not supported by runtime maintainers.
+ - If it falls behind in new features added to the `coantainerd-shim-wasm` due to lack of maintainance
+
+Before removal:
+- We will create an issue in the repository, pinned to the top. 
+- Send notification in our slack channel 
+- make best effort to contact the maintainers (agreed to when adding the shim)
+
+After 1 month of the issue being up, if no maintainer is found we will remove the shim from the project.
```

Comment:
I am intepreting "remove the shim" as deleting the code from the runwasi repo. 

Alternatively, we can archive the not-maintained shim by changing the dependency of `containerd-shim-wasm` crate from a local path to the last released version where the shim is still working. This way we still host the source code of the shim in the main branch and it has visibility to folks checking out the repo, but effectively archiving the shim in a way that it's not going to get the new features from the core crate. 

The downside of that is other dependencies the shim has may or may not be out-of-date and potentially introduce CVEs and as maintainers, we still need to bump the deps for security reasons and that will take time and efforts to maintain.

Reply:
Since we tag commits, the code is affectively archived via git in a way that someone can find it easily. Code is never fully removed. The older releases will also have the built binaries and pointers to the code.

Is the timing here the concern?  Or recognition? something else?  Timing wise, as it is worded now there is 7 months from time we see issue to deleting code.  This is a pretty generous amount of time IMO.

For recognition, maybe we can have a spot in the readme that points to "retired" shims?

Reply:
> maybe we can have a spot in the readme that points to "retired" shims?

I like this

Reply:
I didn't add this explicitly to the doc but I think its a great idea to have a section to recognize older shims.